tests/components/viam/test_config_flow.py

```diff
@@ -0,0 +1,238 @@
+"""Test the viam config flow."""
+
+from collections.abc import Generator
+from unittest.mock import AsyncMock, MagicMock, patch
+
+import pytest
+from viam.app.viam_client import ViamClient
+
+from homeassistant import config_entries
+from homeassistant.components.viam.config_flow import CannotConnect
+from homeassistant.components.viam.const import (
+    CONF_API_ID,
+    CONF_CREDENTIAL_TYPE,
+    CONF_ROBOT,
+    CONF_ROBOT_ID,
+    CONF_SECRET,
+    CRED_TYPE_API_KEY,
+    CRED_TYPE_LOCATION_SECRET,
+    DOMAIN,
+)
+from homeassistant.const import CONF_ADDRESS, CONF_API_KEY
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from .conftest import MockRobot
+
+pytestmark = pytest.mark.usefixtures("mock_setup_entry")
+
+
+async def test_user_form(
+    hass: HomeAssistant,
+    mock_setup_entry: AsyncMock,
+    mock_viam_client: Generator[tuple[MagicMock, MockRobot], None, None],
+) -> None:
+    """Test that the form is served with no input."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    assert result["type"] == FlowResultType.FORM
```

Comment:
```suggestion
    assert result["type"] is FlowResultType.FORM
```
Please update the other occurenses as well